### ISSUE TITLE
Improve semver check for plugins

### ISSUE THREAD

**Issue by mcollina** [fastify/fastify#3882]
Replace the semver check in https://github.com/fastify/fastify/blob/de5c95ee1945ec780a672d580b9e775727b27cc1/lib/pluginUtils.js#L106 with:

```js
const fastifyRc = /-rc.+$/.test(this.version)
if (requiredVersion && semver.satisfies(santizedVersion, requiredVersion, { includePrerelease: fastifyRc }) === false) {
```

https://github.com/npm/node-semver/tree/c56a701f45653940ee8536eafe43b3e46c11d6cc#functions

_Originally posted by @jsumners in https://github.com/fastify/fastify/pull/3879#discussion_r865520807_

**Comment by CynoidIT** [comment]
Would love to help out. What do I need to do to start contributing?

**Comment by CynoidIT** [comment]
Was looking at the code and am not sure why "&& !semver" was changed to "&& semver" in the suggested replacement. Is that intentional or typo?

**Comment by mcollina** [comment]
If all test pass it's ok!

**Comment by jsumners** [comment]
The conditional was made explicit instead of a coercive check.

**Comment by CynoidIT** [comment]
I see. Thanks. PR is up.

**Comment by Fdawgs** [comment]
Closed by #3886
